### PR TITLE
Radek

### PR DESCRIPTION
**Pull request by ridzikowski**
add travels and stop conditions

### CODE REVIEW

File: src/params/travels.jl [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  function (f::Holiday)(state::SimState, params::SimParams, event::Event, infection_time::Real=0.0)
+    @assert kind(event) == OutsideTransmissionEvent
+    ck = contactkind(event)
+    #if ConstantKernelContact !== ck && SporadicContact !== ck && FriendshipContact !== ck
+    #  return true # do not affect other types of contact than "outer" ones
+    #end
+    #rand(state.rng) < max((Float64(-abs(infection_time-f.peak)*abs(infection_time-f.peak)))/1500.0 + f.height,f.minimum)
+    vaule = rand(state.rng)
+    #probability =  max((Float64(-abs(infection_time-f.peak)*abs(infection_time-f.peak)))/1500.0 + f.height,f.minimum)
+    probability =  max((Float64(-abs(event.time-f.peak)*abs(event.time-f.peak)))/1500.0 + f.height,f.minimum)
```
Comment (tozanski) [inline src/params/travels.jl:17]: can you explain how does this function work? what does 1500 stand for?

File: src/simparams.jl [hunk 2]
```diff
@@ -42,10 +44,14 @@ struct SimParams
   phone_tracing_params::Union{Nothing, PhoneTracingParams}
 
   infection_modulation_function::Union{Nothing,InfectionModulation}
+  infection_travels_function::Union{Nothing,InfectionTravels}
+  travels_frequency::TimePoint
```
Comment (tozanski) [inline src/simparams.jl:48]: I believe this can be made a part of the travel function

File: src/event_execution.jl [hunk 3]
```diff
@@ -38,6 +39,24 @@ end
 #
 # transmissions
 #
+function execute!(::Val{OutsideTransmissionEvent}, state::SimState, params::SimParams, event::Event)::Bool
```
Comment (tozanski) [inline src/event_execution.jl:42]: I think we do not need an extra event type for this. `OutsideInfectionEvent` should be sufficient. It is better to have the extra logic in the function that is called at the beginning (something to supersede `initialfeed!` and simply schedule the future infections there

File: src/params/travels.jl [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  end
+  
+  function (f::Holiday)(state::SimState, params::SimParams, event::Event, infection_time::Real=0.0)
+    @assert kind(event) == OutsideTransmissionEvent
+    ck = contactkind(event)
+    #if ConstantKernelContact !== ck && SporadicContact !== ck && FriendshipContact !== ck
+    #  return true # do not affect other types of contact than "outer" ones
+    #end
+    #rand(state.rng) < max((Float64(-abs(infection_time-f.peak)*abs(infection_time-f.peak)))/1500.0 + f.height,f.minimum)
+    vaule = rand(state.rng)
```
Comment (tozanski) [inline src/params/travels.jl:15]: typo

File: src/params/travels.jl [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    #@info "value" vaule
+    #@info "probability" probability
+    vaule < probability
+    
+  end
+  
+  # This all to avoid using @eval and others
+  const travels = Dict{String, Type{T} where T}(
+      "Holiday" => Holiday
+  )
```
Comment (tozanski) [inline src/params/travels.jl:27]: Perhaps the current function `outsidefeed!` can be made in the similar interface and added here as well

File: src/simstate.jl [hunk 6]
```diff
@@ -122,6 +122,9 @@ function initialfeed!(state::SimState, num_initial_infections::Integer, strain::
     push!(state.queue, event)
   end
 
+  event = Event(Val(OutsideTransmissionEvent), 0.0)
```
Comment (tozanski) [inline src/simstate.jl:125]: I think it is better to replace this function with the extended version and allow the used to choose which one should be used

**Review by tozanski — CHANGES_REQUESTED** [review]
Hi, I does look  very good for a first contribution! I've left some comments what I think can be changed. In general the code is good, but I think we must change the design a little bit so that it is easier to extend in the future.

File: src/params/travels.jl (hunk 1 above)
Comment (ridzikowski) [reply src/params/travels.jl:17]: This is an inverted square function whose "peak" can be shifted to choose when you want the highest activity. Additionally, you can define the lower value of the function. 1500 was in the preliminary version. In the correct version there is a "days" factor which determines the length of travel activity.

File: src/params/travels.jl (hunk 4 above)
Comment (ridzikowski) [reply src/params/travels.jl:15]: refactor all travel functions

File: src/simparams.jl (hunk 2 above)
Comment (ridzikowski) [reply src/simparams.jl:48]: ok, i will move it

File: src/event_execution.jl (hunk 3 above)
Comment (ridzikowski) [reply src/event_execution.jl:42]: ok, the current solution is not the worst, but converting it to a function will improve readability.

File: src/simstate.jl (hunk 6 above)
Comment (ridzikowski) [reply src/simstate.jl:125]: if the travel function is not used the event will only be triggered once and will do nothing. The problem will disappear after converting the event `OutsideTransmissionEvent` to a function.

**Pull request closed without merging**